tips/TIP-0038/tip-0038.md

```diff
@@ -0,0 +1,1054 @@
+  </tr>
+</table>
+
+#### Unlocking
+
+The address is unlocked in a transaction if and only if:
+
+- The _Anchor Output_ from whose _Anchor ID_ the address is derived from, is **state transitioned** in the transaction.
+  A governance transition does not unlock the address.
+
+## Signatures
+
+### Ed25519 Signature
+
+The _Ed25519 Signature_ is supported and it is serialized as follows:
+
+<details>
+  <summary>Ed25519 Signature</summary>
+  <blockquote>An Ed25519 Signature with the public key that verifies it.</blockquote>
+</details>
+<table>
+  <tr>
+    <td>
+      <b>Name</b>
+    </td>
+    <td>
+      <b>Type</b>
+    </td>
+    <td>
+      <b>Description</b>
+    </td>
+  </tr>
+  <tr>
+    <td>Signature Type</td>
+    <td>uint8</td>
+    <td>Set to <strong>value 0</strong> to denote an <i>Ed25519 Signature</i>.</td>
+  </tr>
+  <tr>
+    <td>Public Key</td>
+    <td>ByteArray[32]</td>
+    <td>The Ed25519 public key that verifies the signature.</td>
+  </tr>
+  <tr>
+    <td>Signature</td>
+    <td>ByteArray[64]</td>
+    <td>The Ed25519 signature that must be verified according to <a href='../TIP-0014/tip-0014.md'>TIP-14</a>.</td>
+  </tr>
+</table>
+
+## Unlock Conditions
+
+New output features that introduce unlocking conditions, that is, they define constraints on how the output can be
+unlocked and spent, are grouped under the field <i>Unlock Conditions</i>.
+
+Each output **must not contain more than one unlock condition of each type** and not all unlock condition types are
+supported for each output type.
+
+### Address Unlock Condition
+
+It is merely a layout change that the previously defined `Address` field of outputs ([TIP-7](../TIP-0007/tip-0007.md))
+is represented as an <i>Address Unlock Condition</i>. Unlocking an <i>Ed25519 Address</i> doesn't change, it has to be
+performed via a <i>Signature Unlock</i> in a transaction by signing the hash of the transaction essence. Transaction
+validation rules are detailed in [TIP-20](../TIP-0020/tip-0020.md).
+
+New additions are the <i>Account Address</i> and <i>NFT Address</i> types, which have to be unlocked with their
+corresponding unlocks, as defined in [Unlocking Chain Script Locked Outputs](#unlocking-chain-script-locked-outputs),
+and the [Account Output](../TIP-0042/tip-0042.md) and [NFT Output](../TIP-0043/tip-0043.md) TIPs.
+
+<details>
+  <summary>Address Unlock Condition</summary>
+  <blockquote>Defines the Address that owns this output. It can unlock the output with the proper <i>Unlock</i> in a transaction.</blockquote>
+</details>
+<table>
+  <tr>
+    <td>
+      <b>Name</b>
+    </td>
+    <td>
+      <b>Type</b>
+    </td>
+    <td>
+      <b>Description</b>
+    </td>
+  </tr>
+  <tr>
+    <td>Unlock Condition Type</td>
+    <td>uint8</td>
+    <td>Set to <strong>value 0</strong> to denote an <i>Address Unlock Condition</i>.</td>
+  </tr>
+  <tr>
+    <td valign="top">Address <code>oneOf</code></td>
```

Comment:
It is not very clear if something is omitted because it's contextual (like ImplicitAccountCreationAddress only being allowed in Basic outputs) or because it simply is disallowed (like I assume ImplicitAccountCreationAddress is in StorageDepositReturn and Expiration?).

Reply:
So maybe put an explicit note in StorageDepositReturn and Expiration that ImplicitAccountCreationAddress is disallowed or in TIP 42 be a bit more explicit that it can only be used in an AddressUnlockCondition.

Reply:
Ignore me if it's said already somewhere, may have missed it.

Reply:
It's tricky to express this correctly here where Address UC is generically defined, because it suggest that this always is the schema of an Address UC, which is a reasonable assumption given that it's true for every other schema.

Every output currently expresses which address is allowed in its Address UC (Basic allows Impliciti Acc Creation Addresses while others don't).

- One option is to omit the addresses here and adding a note that each output specifies which addresses it allows. 
- Another one is to only include the ones that are always allowed to be present and add a note that each occurence of the Address UC can specify additional ones or something along those lines. Basically how it is now, minus the note. I think this one I like more because it doesn't break the schema definitions as much as the other one.

Reply:
Ok but if we do that, it means that I have to look at ALL tips to see if something is allowed or not. Like, how do I come to the conclusion whether ImplicitAddress is allowed in a StorageDeposit or Expiration? If it just doesn't make sense to start with, I would put a note in this TIP.

Reply:
I'd say you can come to the conclusion that Implicit Account Creation Address it not allowed in the SDRUC because it's not in the SDRUC schema and there's **no note** that occurences of the SDRUC may change that.
Conversely, for Address UC I would add such a note.

I know it's somewhat implicit that way, but it's also not sensible to add a note to every schema that no changes are allowed to it. So in essence, I'd make the changes opt-in by adding a note (like for Addres UC).

Does that sound reasonable?